src/subnet_rental_canister/tests/integration_tests.rs

```diff
@@ -618,17 +625,123 @@ fn test_create_rental_agreement() {
         total_cycles_burned_expected.abs_diff(rental_agreement.total_cycles_created) < TRILLION
     );
 
-    // advance time to after the 180 days
-    pic.advance_time(Duration::from_secs(180 * 86400));
+    // check status of subnet rental
+    let status_ok = check_subnet_status(&pic);
+    assert!(status_ok.description.contains("OK"));
+    assert_eq!(status_ok.days_left, 180 - 1); // since we advaced time by a few minutes above and round down
+
+    // do a topup
+    // set exchange rates
+    let exchange_rate_for_topup = 4_103_000_000; // 1 ICP = 4.103 XDR
+    set_cmc_exchange_rate(&pic, exchange_rate_for_topup);
+    set_xrc_exchange_rate_last_midnight(&pic, exchange_rate_for_topup);
+
+    // get user's ICP balance
+    let user_balance_before_topup = check_balance(&pic, USER_1, DEFAULT_SUBACCOUNT);
+
+    // get estimate for topup
+    let estimate = update_multi_arg::<Result<TopupData, String>>(
+        &pic,
+        SRC_ID,
+        None,
+        "subnet_topup_estimate",
+        (SUBNET_FOR_RENT, Tokens::from_e8s(1_000 * E8S)),
+    )
+    .unwrap()
+    .unwrap();
+
+    println!("estimate: {:?}", estimate);
+
+    // try to do topup with insufficient funds
+    let res = update::<Result<TopupData, String>>(
+        &pic,
+        SRC_ID,
+        Some(USER_1),
+        "process_subnet_topup",
+        SUBNET_FOR_RENT,
+    )
+    .unwrap();
+
+    // the conversion should fail as the user does not have any ICP
+    assert!(res.unwrap_err().contains("insufficient funds"));
```

Comment:
> insufficient funds

Is this just because no ICP was sent? What if you send a small positive amount instead of 0?

Reply:
> as long as it is larger than a default fee

That's what I mean. Instead of sending 0, you could send 9_001 ("OVER NINE THOUSAND!!!!").